File: models/assetmanagement/AssetLocation.js
import mongoose from "mongoose";

const AssetLocationSchema = new mongoose.Schema(
{
    hospital_id: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branch_id: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    building: String,
    floor: String,

    room_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Room"
    },

    description: String

},
{ timestamps: true }
);

export default mongoose.model("AssetLocation", AssetLocationSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AssetLocation.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "asset_locations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AssetLocation extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    private String building;
    private String floor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id")
    private Room room;

    private String description;
}
